src/app/components/pool/action/add/vm-size-picker/vm-size-picker-filter.component.ts

```diff
@@ -0,0 +1,59 @@
+import { ChangeDetectionStrategy, Component, EventEmitter, Input, Output, forwardRef } from "@angular/core";
+import { FormBuilder, FormGroup, NG_VALIDATORS, NG_VALUE_ACCESSOR } from "@angular/forms";
+import { VmSizeFilterValue } from "app/models";
+import { debounceTime, distinctUntilChanged } from "rxjs/operators";
+
+export interface VmSizeFilterCategoryName {
+    label: string;
+    value: string;
+}
+
+@Component({
+    selector: "bl-vm-size-picker-filter",
+    templateUrl: "vm-size-picker-filter.html",
+    providers: [
+        { provide: NG_VALUE_ACCESSOR, useExisting: forwardRef(() => VmSizePickerFilterComponent), multi: true },
+        { provide: NG_VALIDATORS, useExisting: forwardRef(() => VmSizePickerFilterComponent), multi: true },
+    ],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class VmSizePickerFilterComponent {
+    @Input() public set categoriesDisplayName(names: any) {
```

Comment:
use this instead https://stackoverflow.com/questions/45819123/how-to-loop-over-object-properties-with-ngfor-in-angular